### ISSUE TITLE
Crash bug on null number in Duckdb

### ISSUE THREAD

**Issue by lloydtabb** [looker-open-source/malloy#624]
The query below generate NULLs as the result of sums.   I'm getting hard exceptions in DuckDb when running this.

```
terminate called after throwing an instance of 'duckdb::InternalException'
  what():  INTERNAL Error: Calling GetValueInternal on a value that is NULL
Aborted
```

```
        source: state_facts is table('malloytest.state_facts') {
          measure:
            c is airport_count.sum()
            ava is c {where: popular_name = 'Ava'}

          query: basic is {
            group_by: popular_name
            aggregate: c,ava
          }

          query: nested is {
            aggregate: c
            nest: basic is {
              group_by: popular_name
              aggregate: c,ava
            }
          }
        }
        query: state_facts->nested
```

**Comment by lloydtabb** [comment]
Here is a test case that will fail for nomodel.spec.ts

```
  it(`number as null- ${databaseName}`, async () => {
    // a cross join produces a Many to Many result.
    // symmetric aggregate are needed on both sides of the join
    // Check the row count and that sums on each side work properly.
    const result = await runtime
      .loadQuery(
        `
        source: s is table('malloytest.state_facts') + {
        }
        query: s-> {
          group_by: state
          nest: ugly is {
            group_by: popular_name
            aggregate: foo is NULLIF(sum(airport_count)*0,0)+1
          }
        }
      `
      )
      .run();
    expect(result.data.path(0, "ugly", 0, "foo").value).toBe(null);
  });
  ```

**Comment by lloydtabb** [comment]
erros with
```
terminate called after throwing an instance of 'duckdb::InternalException'
  what():  INTERNAL Error: Calling GetValueInternal on a value that is NULL
```

**Comment by whscullin** [comment]
The fix has landed in duckdb, it should be available later this evening so I will probably bump duckdb in the morning.

**Comment by whscullin** [comment]
Fixed by #667
